### ISSUE TITLE
Outline tests needed to be performed on sky before the eclipse

### ISSUE THREAD

**Issue by bberkeyU** [NCAR/ucomp-pipeline#246]
On sky tests with UCoMP before the eclipse. This should cover all tests we expect we need to finish and have validated before the March 28/31 fights. (Hopefully March 28/31 have no engineering.)

We should add a plan for how long each test should take.  And maybe build some tools to process files, etc, if any decisions need to be made on the mountain.

- Find the center line for each line we plan to observe.  Which lines?
- Polarization calibration (take de-modulation data)? Which lines?

**Comment by detoma** [comment]
On November 25, 2022, the last day before the eruption that science data were taken at MLSO, the new filters were roughly centered on the line. Steve performed a series of wide scans to find the line center tuning. The last wide scan, in steps of 0.07nm was done on November 24 2022, the penultimate day before closings, for the 802nm line. I have not checked if the wavelength calibration needed fine tuning. I am not aware of any post-processing done to check how well the lines were centered, but Steve may have done it. Also, I have not seen scans with small wavelength steps. The 3 points final scans on November 25, 2022 have steps of 0.09nm for 991, 0.06nm step for 802 and 761, and 0.05nm steps for, 670. It is possible that some fine tuning was still needed for the new filters. 

In any case, we do not know what the status of the instrument is now and if wavelengths have drifted during the 1+ year MLSO has been closed, so the wavelength must be recalibrated for all filters. For the first day we can take data, I asked to perform a series of wide scans to find line center position for all filters, old or new, that we plan to observe with during the eclipse. Ben suggested to use the observed wavelength drift before closure to guess where the lines' center position is now. The scans should be wide enough to include the nominal position of the line at the time of closing and the projected one. The steps separation for each lines are TBD.

If time allows, a polarimetric calibration would be helpful to verify that the demodulation matrix has not changed. This requires to be able to run Steve's code to compute the demodulation matrix.

**Comment by bberkeyU** [comment]
To find the wavelength offsets (in the bright lines) I would like to perform 2 passes.  We assume the drift filter remained constant last year for the first pass. For 1074 this would be a .23nm/year over 1.3 years =0.3nm drift.  We would then scan from this new position +/- .36nm looking for the brightest coronal feature.  We can do this scan in 7 points (with a .12nm spacing slightly under the fwhm of the filter); which should take 90 seconds for UCoMP to collect.  And do the analysis by a visual inspection of the L0 in FV or…?  In 1074 at least, it is very obvious which is the peak wavelength looking at just the onband L0 data.  And should take under 10 minutes to verify.  I need to check this for the other lines; to see if it is possible. 


We would then recompute the offset by adding the wavelength of the peak extension -1074.7

At this point, we would run a second pass with a more granular stepsize between wavelengths taking appropriate flats and darks.  Since we are close to the correct wavelength this could be accomplished with a 5-point scan.  We will need to decide how many repeats to run 1 ala waves or 4 ala synoptic (maybe 1 repeat for 1074, 4 for 637, or a similar scheme.   

If we can perform the first pass for all 5 wavelengths ahead of time.  We would need an additional hour to run coronal and flats with 7 wavelengths across the line and 4 repeats.  This should give us enough data to completely refine the wavelength tuning.  If we wanted to consider reducing this time we could reduce the repeats or go to 5 points across the line.


However, if we are uncertain/unable to do the first pass for a wave region due to SNR issues for L0 inspection, we either need to develop some better tools for real-time analysis (outside the pipeline) or consider long scans.  If we scanned the whole 637 region with tunings spaced by a fwhm we would need 25 scans across the line (+25 more for the flats).  706 would need 19 and 789 20.   We would expect these 3 scans (coronal + flats) to take about 107minutes and give fairly poor wavelength coverage.  We also couldn't run the detailed scan on the same trip as we would rely on offline processing by the pipeline to fine the peak wavelength.   I think we should try to avoid scanning the whole prefilter.

**Comment by bberkeyU** [comment]
Polarization calibrations take about 4 minutes per wavelength.

**Comment by bberkeyU** [comment]
I have pulled the ucomp wavelength offsets out of our git history and started looking at the relationship to have the offsets change vs wave region with time.

The drift speed we measured varied linearly with wavelength.  To first order, even if the drift speed in 2023 slowed from 2022, this linear relationship between the speed at the various wave regions will still hold.  If we can convince ourselves that this is true, we may only need to do the first pass scan at 1074 and then apply a derated drift speed to the other lines, which is a starting point for the second pass.

Notebook in:
https://github.com/NCAR/ucomp-studies/blob/master/Lyot-filter-drift/measured-drive-vs-wave.ipynb

**Comment by bberkeyU** [comment]
Starting to put together eclipse prep recipes.  To cover the emission peak scans, distortion, and polarization calibration scans.

For this, I am planning to put together one large menu file full of smaller cookbooks. Until the eclipse prep is done, we will run programs from this menu.  As currently written (3/14/2024), we have 4 hours of data collection in the menu file.   We should expect the best case to get through about 2 hours per day.    


https://github.com/NCAR/ucomp-configuration/blob/main/Recipes/eclipsePrep.md

**Comment by bberkeyU** [comment]
Recipes updated to match our Day 1 plans.   The github recipe/menu/cookbook markdown file creation code was updated to add colored icons to denote data type of the observations.  This should help us make quicker visual inspections to see if we are getting the measurements we expect (corona data, darks, flats).


Below are the captured notes from the Day 1 doc.
1074 narrow scan 7 points 4 repeats 16 sums, spacing = 0.06 nm
1079  narrow scan 7 points 4 repeats 16sums, spacing = 0.06 nm   
 (visual check of data)
789    narrow scan 9 points 4 repeats 16sums, spacing = 0.05 nm
637    narrow scan 9 points 4 repeats 16sums, spacing = 0.04 nm
706    narrow scan 9 points 4 repeats 16sums, spacing = 0.04 nm

991   narrow scan 9 points 4 repeats 16sums, spacing = 0.06 nm
802   narrow scan 9 points 4 repeats 16sums, spacing = 0.05 nm
761  narrow scan 9 points 4 repeats 16sums, spacing = 0.05 nm
670  narrow scan 9 points 4 repeats 16sums, spacing = 0.04 nm

**Comment by bberkeyU** [comment]
New polarization datasets were captured for 637, 789,1074,1079. At this point, they may not have been processed.

On March 30, we performed the scans defined above (with wavoff +.30nm from the November 2022 data).  We found no drift in the data, and visual inspection of the peak in the tuning found the best wavoff at the November 2022 position.    Small stepsize data was taken for 637,706,789,1074,1079 to compute the best wavoff.

This was a major engineering result; the Lyot filter wavelength offset appears only when the filter is being used.  What we dont know is if this drift is related to heating of the filter to 35C or tuning of the LCVRs.
